build failure

I do `./configure --disable-avs --enable-lto   --disable-ffms --disable-swscale --disable-lavf  --disable-gpac  --disable-lsmash` and `make -j` on macOS ,but it fails.

```
x264.c:1774:2: error: function definition is not allowed here
        {
        ^
x264.c:1820:19: error: call to undeclared function 'StringPadRight'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
    x264_cli_log( StringPadRight(demuxername, padLen, " "), X264_LOG_INFO, "%dx%d %u:%u @ %u/%u fps (%cfr)\n", info.width,
                  ^
x264.c:1820:19: error: incompatible integer to pointer conversion passing 'int' to parameter of type 'const char *' [-Wint-conversion]
    x264_cli_log( StringPadRight(demuxername, padLen, " "), X264_LOG_INFO, "%dx%d %u:%u @ %u/%u fps (%cfr)\n", info.width,
                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
x264.c:284:32: note: passing argument to parameter 'name' here
void x264_cli_log( const char *name, int i_level, const char *fmt, ... )
                               ^
x264.c:1823:19: error: incompatible integer to pointer conversion passing 'int' to parameter of type 'const char *' [-Wint-conversion]gcc -Wshadow -O3 -ffast-math  -Wall -I. -I. -mdynamic-no-pic -arch arm64 -std=gnu99 -D_GNU_SOURCE -fomit-frame-pointer -flto -fno-tree-vectorize -fvisibility=hidden   -c common/quant.c -o common/quant-8.o -DHIGH_BIT_DEPTH=0 -DBIT_DEPTH=8

    x264_cli_log( StringPadRight(demuxername, padLen, " "), X264_LOG_INFO, "color matrix: %s\n", strtable_lookup(x264_colmatrix_names, shown_colormatrix) );
                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
x264.c:284:32: note: passing argument to parameter 'name' here
void x264_cli_log( const char *name, int i_level, const char *fmt, ... )
```

this part still exists?

```
x264.c:1774:2: error: function definition is not allowed here
        {
        ^
x264.c:1820:19: error: call to undeclared function 'StringPadRight'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
    x264_cli_log( StringPadRight(demuxername, padLen, " "), X264_LOG_INFO, "%dx%d %u:%u @ %u/%u fps (%cfr)\n", info.width,
                  ^
x264.c:1820:19: error: incompatible integer to pointer conversion passing 'int' to parameter of type 'const char *' [-Wint-conversion]
    x264_cli_log( StringPadRight(demuxername, padLen, " "), X264_LOG_INFO, "%dx%d %u:%u @ %u/%u fps (%cfr)\n", info.width,
                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
x264.c:284:32: note: passing argument to parameter 'name' here
void x264_cli_log( const char *name, int i_level, const char *fmt, ... )
                               ^
x264.c:1823:19: error: incompatible integer to pointer conversion passing 'int' to parameter of type 'const char *' [-Wint-conversion]gcc -Wshadow -O3 -ffast-math  -Wall -I. -I. -mdynamic-no-pic -arch arm64 -std=gnu99 -D_GNU_SOURCE -fomit-frame-pointer -flto -fno-tree-vectorize -fvisibility=hidden   -c common/quant.c -o common/quant-8.o -DHIGH_BIT_DEPTH=0 -DBIT_DEPTH=8

    x264_cli_log( StringPadRight(demuxername, padLen, " "), X264_LOG_INFO, "color matrix: %s\n", strtable_lookup(x264_colmatrix_names, shown_colormatrix) );
                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
x264.c:284:32: note: passing argument to parameter 'name' here
void x264_cli_log( const char *name, int i_level, const char *fmt, ... )
```

> this part still exists?
> 
> ```
> x264.c:1774:2: error: function definition is not allowed here
>         {
>         ^
> x264.c:1820:19: error: call to undeclared function 'StringPadRight'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
>     x264_cli_log( StringPadRight(demuxername, padLen, " "), X264_LOG_INFO, "%dx%d %u:%u @ %u/%u fps (%cfr)\n", info.width,
>                   ^
> x264.c:1820:19: error: incompatible integer to pointer conversion passing 'int' to parameter of type 'const char *' [-Wint-conversion]
>     x264_cli_log( StringPadRight(demuxername, padLen, " "), X264_LOG_INFO, "%dx%d %u:%u @ %u/%u fps (%cfr)\n", info.width,
>                   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
> x264.c:284:32: note: passing argument to parameter 'name' here
> void x264_cli_log( const char *name, int i_level, const char *fmt, ... )
>                                ^
> x264.c:1823:19: error: incompatible integer to pointer conversion passing 'int' to parameter of type 'const char *' [-Wint-conversion]gcc -Wshadow -O3 -ffast-math  -Wall -I. -I. -mdynamic-no-pic -arch arm64 -std=gnu99 -D_GNU_SOURCE -fomit-frame-pointer -flto -fno-tree-vectorize -fvisibility=hidden   -c common/quant.c -o common/quant-8.o -DHIGH_BIT_DEPTH=0 -DBIT_DEPTH=8
> 
>     x264_cli_log( StringPadRight(demuxername, padLen, " "), X264_LOG_INFO, "color matrix: %s\n", strtable_lookup(x264_colmatrix_names, shown_colormatrix) );
>                   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
> x264.c:284:32: note: passing argument to parameter 'name' here
> void x264_cli_log( const char *name, int i_level, const char *fmt, ... )
> ```

yeah. I will try if I can reproduce on github CI.

--- a/x264.c
+++ b/x264.c
@@ -1501,6 +1501,21 @@ static int parse_enum_value( const char *arg, const char * const *names, int *ds
     return -1;
 }
 
+char *StringPadRight(char *string, int padded_len, char *pad)
+{
+    int len = (int) strlen(string);
+    if (len >= padded_len)
+	{
+        return string;
+    }
+    int i;
+    for (i = 0; i < padded_len - len; i++)
+	{
+        strcat(string, pad);
+    }
+    return string;
+}
+
 static int parse( int argc, char **argv, x264_param_t *param, cli_opt_t *opt )
 {
     char *input_filename = NULL;
@@ -1769,21 +1784,6 @@ generic_option:
     video_info_t info = {0};
     char demuxername[5];
     int padLen = 4;
-
-	char *StringPadRight(char *string, int padded_len, char *pad)
-	{
-    int len = (int) strlen(string);
-    if (len >= padded_len)
-	{
-        return string;
-    }
-    int i;
-    for (i = 0; i < padded_len - len; i++)
-	{
-        strcat(string, pad);
-    }
-    return string;
-    }
 
     /* set info flags to be overwritten by demuxer as necessary. */
     info.colormatrix = param->vui.i_colmatrix;

